### PR TITLE
Benchmarking Workflow

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

**Comment by amitksingh1490** [comment]
[0m Running Warmup plaintext [0m wrk -H 'Host: tfb-server' -H 'Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7' -H 'Connection: keep-alive' --latency -d 15 -c 512 --timeout 8 -t 2 http://tfb-server:8080/plaintext [0m--------------------------------------------------------- [0m[0mRunning 15s test @ http://tfb-server:8080/plaintext [0m 2 threads and 512 connections [0m Thread Stats Avg Stdev Max +/- Stdev [0m Latency 32.02ms 138.05ms 1.26s 96.00% [0m Req/Sec 25.61k 6.06k 49.65k 83.10% [0m Latency Distribution [0m 50% 5.62ms [0m 75% 8.31ms [0m 90% 12.60ms [0m 99% 890.90ms [0m 757446 requests in 15.10s, 96.07MB read [0mRequests/sec: 50178.04 [0mTransfer/sec: 6.36MB [0m[0m--------------------------------------------------------- [0m Concurrency: 256 for plaintext [0m wrk -H 'Host: tfb-server' -H 'Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7' -H 'Connection: keep-alive' --latency -d 15 -c 256 --timeout 8 -t 2 http://tfb-server:8080/plaintext -s pipeline.lua -- 16

**Comment by amitksingh1490** [comment]
[0m[0m--------------------------------------------------------- [0m Concurrency: 256 for plaintext [0m wrk -H 'Host: tfb-server' -H 'Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7' -H 'Connection: keep-alive' --latency -d 15 -c 256 --timeout 8 -t 2 http://tfb-server:8080/plaintext -s pipeline.lua -- 16 [0m--------------------------------------------------------- [0m[0mRunning 15s test @ http://tfb-server:8080/plaintext [0m 2 threads and 256 connections [0m Thread Stats Avg Stdev Max +/- Stdev [0m Latency 57.43ms 236.36ms 2.53s 95.28% [0m Req/Sec 134.66k 32.04k 192.81k 62.67% [0m Latency Distribution [0m 50% 8.62ms [0m 75% 13.35ms [0m 90% 21.28ms [0m 99% 1.42s [0m 4019648 requests in 15.04s, 509.85MB read [0mRequests/sec: 267319.48 [0mTransfer/sec: 33.91MB [0mSTARTTIME 1637212672 [0mENDTIME 1637212687

**Comment by amitksingh1490** [comment]
[0mRequests/sec: 258631.02

**Comment by amitksingh1490** [comment]
[0mRequests/sec: 46156.83

**Comment by amitksingh1490** [comment]
``````

**Comment by amitksingh1490** [comment]
```[0m[0m[0m--------------------------------------------------------- [0m Concurrency: 256 for plaintext [0m wrk -H 'Host: tfb-server' -H 'Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7' -H 'Connection: keep-alive' --latency -d 15 -c 256 --timeout 8 -t 2 http://tfb-server:8080/plaintext -s pipeline.lua -- 16 [0m--------------------------------------------------------- [0m[0mRunning 15s test @ http://tfb-server:8080/plaintext [0m 2 threads and 256 connections [0m Thread Stats Avg Stdev Max +/- Stdev [0m Latency 31.14ms 18.49ms 210.80ms 65.98% [0m Req/Sec 38.47k 3.99k 49.83k 78.00% [0m Latency Distribution [0m 50% 29.24ms [0m 75% 43.73ms [0m 90% 54.99ms [0m 99% 80.67ms [0m 1148907 requests in 15.04s, 145.73MB read [0mRequests/sec: 76390.76 [0mTransfer/sec: 9.69MB [0m[2mzio-http: [0mNov 18, 2021 8:56:36 AM io.netty.channel.DefaultChannelPipeline onUnhandledInboundException [0m[2mzio-http: [0mWARNING: An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.```

**Comment by amitksingh1490** [comment]
``````

**Comment by amitksingh1490** [comment]
``````

**Comment by amitksingh1490** [comment]
``````

**Comment by amitksingh1490** [comment]
```Requests/sec: 67461.72```

**Comment by amitksingh1490** [comment]
```Requests/sec: 69272.06```

**Comment by amitksingh1490** [comment]
#### Benchmark Results ##### Concurrency: Concurrency: 256 Dockerfile LICENSE README.md deployment frameworks package.json result results tfb toolset Requests/sec: 47737.31 🚀

**Comment by amitksingh1490** [comment]
Benchmark Results For Concurrency: 256: Requests/sec: 70508.14

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 63465.55**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 47082.55**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 67781.54**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 43479.09**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 401308.54**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 404807.30**

File: project/BenchmarkWorkFlow.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      name = "Benchmarks",
+      oses = List("centos"),
+      cond = Some("${{ github.event_name == 'push'}}"),
+      env = Map("GITHUB_TOKEN" -> "${{secrets.ACTIONS_PAT}}"),
+      steps = List(
+        WorkflowStep.Checkout,
+        WorkflowStep.Use(
+          UseRef.Public("actions", "checkout", s"v2"),
+          Map(
+            "repository" -> "amitksingh1490/FrameworkBenchmarks",
```
Comment (tusharmath) [inline project/BenchmarkWorkFlow.scala]:
```suggestion
            "repository" -> "dream11/FrameworkBenchmarks",
```
Either this ... or can we do something else?

**Comment by tusharmath** [comment]
how long does the benchmarking test take? Can we run it on every commit?

**Comment by amitksingh1490** [comment]
> how long does the benchmarking test take? Can we run it on every commit?

It takes around 2 min 15 secs.

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 404125.78**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 402560.49**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 408082.10**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 404792.92**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 407695.91**

**Pull request merged**